exonum-java-proofs/src/test/java/com/exonum/binding/storage/proofs/map/flat/CheckedFlatMapProofTest.java

```diff
@@ -0,0 +1,123 @@
+package com.exonum.binding.storage.proofs.map.flat;
+
+import static com.exonum.binding.hash.Funnels.hashCodeFunnel;
+import static com.exonum.binding.storage.proofs.DbKeyFunnel.dbKeyFunnel;
+import static com.exonum.binding.test.Bytes.bytes;
+import static com.exonum.binding.test.Bytes.createPrefixed;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.exonum.binding.hash.HashCode;
+import com.exonum.binding.hash.HashFunction;
+import com.exonum.binding.hash.Hashing;
+import com.exonum.binding.storage.proofs.map.DbKey;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Test;
+
+public class CheckedFlatMapProofTest {
+
+  private static final byte[] VALUE = "testValue".getBytes();
+  private static final HashFunction HASH_FUNCTION = Hashing.defaultHashFunction();
+
+  @Test
+  public void mapProofShouldBeValid() {
+    byte[] firstKey = createPrefixed(bytes(0b001101), DbKey.KEY_SIZE);
+    byte[] secondKey = createPrefixed(bytes(0b011101), DbKey.KEY_SIZE);
+    byte[] thirdKey = createPrefixed(bytes(0b1111101), DbKey.KEY_SIZE);
+
+    DbKey valueKey = DbKey.newLeafKey(secondKey);
+    MapProofEntryLeaf leaf = createLeafMapEntry(valueKey.getKeySlice(), VALUE);
+    List<MapProofEntry> entries = Arrays.asList(
+        createBranchMapEntry(firstKey),
+        leaf,
+        createBranchMapEntry(thirdKey)
+    );
+    UncheckedMapProof uncheckedFlatMapProof = new UncheckedFlatMapProof(entries);
+
+    CheckedMapProof checkedMapProof = uncheckedFlatMapProof.check();
+    assertThat(entries, equalTo(uncheckedFlatMapProof.getProofList()));
+
+    CheckedMapProofEntry checkedEntry = new CheckedMapProofEntry(valueKey.getKeySlice(), VALUE);
+    assertThat(Collections.singletonList(checkedEntry), equalTo(checkedMapProof.getEntries()));
+    assertTrue(checkedMapProof.containsKey(valueKey.getKeySlice()));
```

Comment:
There are four places where valueKey.getKeySlice() is used (essentially, that's the `secondKey`). Every time it forces the reader to think 'what is a key slice in this case'. What if we rename the `secondKey` to `valueKey`, and present `valueKey` to `valueDbKey` to avoid confusion?